vmm/src/memory_manager.rs

```diff
@@ -140,12 +140,13 @@ pub type MemoryZones = HashMap<String, MemoryZone>;
 
 #[derive(Clone, Serialize, Deserialize, Versionize)]
 struct GuestRamMapping {
-    slot: u32,
+    slot: Option<u32>,
     gpa: u64,
     size: u64,
     zone_id: String,
     virtio_mem: bool,
     file_offset: u64,
+    host_addr: u64,
```

Comment:
I think reusing the `GuestRamMapping` struct is not a good idea. This struct was created for getting dirty pages for VM migration via tracking calls to and returns of `create_userspace_mapping()`, which is why a new `GuestRamMapping` entry is created if and only if there is a call to `create_userspace_mapping()`. On the other hand, this PR tries to split calls to `create_userspace_mapping()` from `MemoryManager::new()` while not splitting the creation of new  `GuestRamMapping` entries. This is why making `slot` optional is quite award in terms of requiring manual updates of its value and `unwrap()` calls. 

I think it would be cleaner if we create a new struct (that includes gpa, size and host_addr) to serve the purpose of the PR while keeping `GuestRamMapping` struct unchanged.

Reply:
How about we move the call to `allocate_address_space()` right before `start_boot_vcpus()`? In that way we don't need to create eager GuestRamMappings? @likebreath

Reply:
@jinankjain I like this idea. The patch now is much cleaner. 

But looks like the changes broke the snapshot/restore on aarch64? If you need inputs for debugging, @michael2012z and @MrXinWang are the experts in this domain.

Reply:
Yeah, I would appreciate that help since I don't have the ARM hardware to debug this problem.

Reply:
Sorry, I added some comments a few days ago but I forgot to submit the review.

Please see my comments. And there is a suggested change to resolve the test failure.